### PR TITLE
Psycopg autocommit handling

### PR DESCRIPTION
**Pull request by elpavel**
##### SUMMARY
Improved autocommit handling, to work with Psycopg 3 when we add the support.
It will help with #499

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
postgresql_tablespace

##### ADDITIONAL INFORMATION

The following doesn't work in Psycopg 3
```
connection.set_session(autocommit=True)
```

it is being replaced with
```
connection.autocommit = True
```
both methods were introduced in psycopg 2.4.2 https://www.psycopg.org/docs/connection.html#connection.autocommit

The code has been tested with Psycopg 3 in #517

### CODE REVIEW

File: plugins/modules/postgresql_tablespace.py [hunk 1]
```diff
@@ -437,8 +437,8 @@ def main():
 
     # Change autocommit to False if check_mode:
     if module.check_mode:
-        if PSYCOPG2_VERSION >= '2.4.2':
-            db_connection.set_session(autocommit=False)
+        if PSYCOPG_VERSION >= LooseVersion("2.4.2"):
+            db_connection.autocommit = False
```
Comment (Andersson007) [inline plugins/modules/postgresql_tablespace.py:441]:
```suggestion
        if PSYCOPG_VERSION >= LooseVersion("2.4.2"):
            db_connection.autocommit = False
```
@elpavel as far as i can see `connect_to_db`  now cares about all that stuff. Should we maybe change the line a bit above:
```
db_connection, dummy = connect_to_db(module, conn_params, autocommit=True)
```
to something like:
```
db_connection, dummy = connect_to_db(module, conn_params, autocommit=True if not module.check_mode else False)
```
I'm not sure if this exact syntax works though, but the idea should be clear

Comment (Andersson007) [reply plugins/modules/postgresql_tablespace.py:441]: ah, i see now that it repeats:) so maybe also a separate function just to reduce code repetition?

File: changelogs/fragments/533-postgresql_tablespace.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- "postgresql_tablespace - improved autocommit handling (https://github.com/ansible-collections/community.postgresql/pull/533)."
```
Comment (Andersson007) [inline changelogs/fragments/533-postgresql_tablespace.yml:2]:
```suggestion
minor_changes:
- "postgresql_tablespace - improved autocommit handling (https://github.com/ansible-collections/community.postgresql/pull/533)."
```
as this is just refactoring, could you please remove the fragment?

File: plugins/modules/postgresql_tablespace.py (hunk 1 above)
Comment (Andersson007) [reply plugins/modules/postgresql_tablespace.py:441]: I'm not insisting. If you think that now is better, please just mark this as solved

**Review by Andersson007 — APPROVED** [review]

File: plugins/modules/postgresql_tablespace.py (hunk 1 above)
Comment (elpavel) [reply plugins/modules/postgresql_tablespace.py:441]: Separate function would make sense, as we use the same logic three times. The function could be moved to postgres.py, to keep the psycopg logic there.

Comment (Andersson007) [reply plugins/modules/postgresql_tablespace.py:441]: would you like to do it here or later (which would be also fine) ?

Comment (elpavel) [reply plugins/modules/postgresql_tablespace.py:441]: I will do it here.

**Comment by elpavel** [comment]
Sanity checks failing for 2.16 with
```
Running sanity test "ignores"
ERROR: Found 1 ignores issue(s) which need to be resolved:
ERROR: tests/sanity/ignore-2.16.txt:1:1: Sanity test 'replace-urlopen' does not test path 'tests/utils/shippable/check_matrix.py' (0%)
See documentation for help: https://docs.ansible.com/ansible-core/devel/dev_guide/testing/sanity/ignores.html
```
any ideas what this means?

**Comment by Andersson007** [comment]
> Sanity checks failing for 2.16 with
> 
> ```
> Running sanity test "ignores"
> ERROR: Found 1 ignores issue(s) which need to be resolved:
> ERROR: tests/sanity/ignore-2.16.txt:1:1: Sanity test 'replace-urlopen' does not test path 'tests/utils/shippable/check_matrix.py' (0%)
> See documentation for help: https://docs.ansible.com/ansible-core/devel/dev_guide/testing/sanity/ignores.html
> ```
> 
> any ideas what this means?

@elpavel i commented in that PR about urlopen, thanks!

**Review by jchancojr — APPROVED** [review]

**Review by hunleyd — APPROVED** [review]

**Pull request merged**